sdk/communication/AzureCommunication/Source/Authentication/CommunicationTokenCredentialProtocol.swift

```diff
@@ -0,0 +1,41 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+#if canImport(AzureCore)
+import AzureCore
+#endif
+import Foundation
+/**
+ Protocol defining the shape of credentials used with Azure Communication Services.
+ */
+public protocol CommunicationTokenCredentialProtocol {
```

Comment:
This used to be named `CommunicationTokenCredential` so I had to rename it. 
I personally am not a fan of just appending `Protocol` to the end. If you guys have suggestion for better name please advise.

Reply:
How about "CommunicationTokenProvider"? Or perhaps this might be an example of a "verb" protocol, e.g. "CommunicationTokenProviding"?

Also might it be a good idea to align the names of the credential / protocol, and the callback type? The callback type is currently just called "AccessTokenRefreshOnCompletion". Perhaps something like "CommuncationTokenCompletionHandler" would be more clear.

Reply:
I like the idea of using a verb.